src/session.ts

```diff
@@ -68,21 +72,18 @@ async function encryptSession(session: Session) {
 
 async function withAuth(
   request: Request,
-  options?: {
-    ensureSignedIn?: false;
+  options: {
+    ensureSignedIn?: boolean;
     debug?: boolean;
+    data?: object;
+    headers?: Record<string, string>;
```

Comment:
Should we split up our options object from headers and pass it as a third argument of type `ResponseInit`? This would also allow users to provide a response status.

Reply:
Also nit: any reason for the overload here and not typing the implementation?

Reply:
Legacy code from how things worked in the Next.js implementation, will remove it.